diff -Naur polyorb-gpl-2011-src/compilers/iac/backend-be_corba_ada-nutils.adb polyorb-gpl-2011-iac/compilers/iac/backend-be_corba_ada-nutils.adb
--- polyorb-gpl-2011-src/compilers/iac/backend-be_corba_ada-nutils.adb	2011-04-19 13:01:59.000000000 +0300
+++ polyorb-gpl-2011-iac/compilers/iac/backend-be_corba_ada-nutils.adb	2011-08-18 18:08:40.000000000 +0300
@@ -95,6 +95,10 @@
    The_Unique_Suffix : access constant String := Latin_1_Unique_Suffix'Access;
    The_Unique_Infix  : access constant String := Latin_1_Unique_Infix'Access;
 
+   --  argument for pragma Wide_Character_Encoding
+   Encoding_Name  : Name_Id;  --  Brackets or UTF8
+   Use_UTF8       : Boolean := False;
+
    procedure New_Operator (Op : Operator_Type; I : String := "");
 
    function Internal_Name (P : Node_Id; L : GLists) return Name_Id;
@@ -1073,6 +1077,14 @@
 
       Set_Str_To_Name_Buffer ("DomainManager");
       DomainManager_Name := Name_Find;
+
+      if Use_UTF8 then
+         Set_Str_To_Name_Buffer ("UTF8");
+      else
+         Set_Str_To_Name_Buffer ("Brackets");
+      end if;
+
+      Encoding_Name := Name_Find;
    end Initialize;
 
    --------------
@@ -1750,6 +1762,9 @@
       function Make_Style_Check_Pragma return Node_Id;
       --  Returns a node for a pragma deactivating style checks
 
+      function Make_Character_Encoding_Pragma return Node_Id;
+      --  Returns a node for an encoding specification pragma
+
       -----------------------------
       -- Make_Style_Check_Pragma --
       -----------------------------
@@ -1769,6 +1784,17 @@
                         (New_String_Value (Name_Find, Wide => False))));
       end Make_Style_Check_Pragma;
 
+      ------------------------------------
+      -- Make_Character_Encoding_Pragma --
+      ------------------------------------
+
+      function Make_Character_Encoding_Pragma return Node_Id is
+      begin
+         return Make_Pragma
+                  (Pragma_Wide_Character_Encoding,
+                   New_List (Make_Identifier (Encoding_Name)));
+      end Make_Character_Encoding_Pragma;
+
    --  Start of processing for Make_Package_Declaration
 
    begin
@@ -1805,6 +1831,7 @@
       --  comment because some lines in the header may be very long.
 
       Append_To (Context_Clause (Pkg), Make_Style_Check_Pragma);
+      Append_To (Context_Clause (Pkg), Make_Character_Encoding_Pragma);
 
       --  Adding a comment header
 
@@ -2849,6 +2876,7 @@
    begin
       The_Unique_Suffix := UTF_8_Unique_Suffix'Access;
       The_Unique_Infix  := UTF_8_Unique_Infix'Access;
+      Use_UTF8          := True;
    end Set_UTF_8_Encoding;
 
 end Backend.BE_CORBA_Ada.Nutils;
diff -Naur polyorb-gpl-2011-src/compilers/iac/backend-be_corba_ada-nutils.ads polyorb-gpl-2011-iac/compilers/iac/backend-be_corba_ada-nutils.ads
--- polyorb-gpl-2011-src/compilers/iac/backend-be_corba_ada-nutils.ads	2011-04-19 13:01:59.000000000 +0300
+++ polyorb-gpl-2011-iac/compilers/iac/backend-be_corba_ada-nutils.ads	2011-08-18 17:47:27.000000000 +0300
@@ -467,7 +467,8 @@
       Pragma_Style_Checks,
       Pragma_Suppress,
       Pragma_Unreferenced,
-      Pragma_Warnings);
+      Pragma_Warnings,
+      Pragma_Wide_Character_Encoding);
 
    GN : array (Pragma_Id) of Name_Id;
    --  Array of pragma identifiers
