diff -Naur gcc-4.3.6/gcc/ada/sem_ch3.adb gcc-4.3.6_/gcc/ada/sem_ch3.adb
--- gcc-4.3.6/gcc/ada/sem_ch3.adb	2010-06-03 09:27:29.000000000 +0300
+++ gcc-4.3.6_/gcc/ada/sem_ch3.adb	2011-06-01 14:57:59.000000000 +0300
@@ -12865,7 +12865,7 @@
                      Type_Conformant (Subp, Act_Subp,
                                       Skip_Controlling_Formals => True)))
             then
-               pragma Assert (not Is_Ancestor (Parent_Base, Generic_Actual));
+               --  pragmaAssert(not Is_Ancestor (Parent_Base, Generic_Actual));
 
                --  Remember that we need searching for all pending primitives
 
