diff -Naur gprbuild-2010-src/src/gprlib.adb gprbuild-2010-src_/src/gprlib.adb
--- gprbuild-2010-src/src/gprlib.adb	2010-06-08 18:33:46.000000000 +0300
+++ gprbuild-2010-src_/src/gprlib.adb	2011-06-01 17:00:21.000000000 +0300
@@ -1055,7 +1055,13 @@
                      else
                         Compiler_Name := new String'(Line (1 .. Last));
 
-                        if Last > 3
+                        if Last > 7
+                          and then Line (Last - 6 .. Last) = "gnatgcc"
+                        then
+                           Gnatbind_Name :=
+                             new String'(Line (1 .. Last - 7) & "gnatbind");
+
+                        elsif Last > 3
                           and then Line (Last - 2 .. Last) = "gcc"
                         then
                            Gnatbind_Name :=
--- gprbuild-2010-src/share/gprconfig/linker.xml	2011-03-16 19:50:00.000000000 +0200
+++ gprbuild-2010-src_/share/gprconfig/linker.xml	2011-07-05 16:59:40.000000000 +0300
@@ -366,6 +366,9 @@
     <targets>
        <target name="^.*linux.*$" />
     </targets>
+   <compilers negate="true">
+     <compiler name="GNAT-Gentoo" />
+   </compilers>
     <config>
    for Library_Partial_Linker use ("${PATH(ada)}${PREFIX(ada)}gcc", "-nostdlib", "-Wl,-r", "-o");
    for Run_Path_Option  use ("-Wl,-z,origin,-rpath,");
